hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -679,6 +680,17 @@ protected TcpClientConnection createSocketConnection(Address target) {
         }
     }
 
+    private Address translate(Member member) {
+        if (client.getClientClusterService().translateToPublicAddress()) {
+            Address publicAddress = member.getAddressMap().get(EndpointQualifier.resolve(ProtocolType.CLIENT, "public"));
```

Comment:
nit: `EndpointQualifier.resolve(ProtocolType.CLIENT, "public")` creates a new object on each call. It could be avoided with a `static field` const. The same consideration applies to similar server-side usages.